### ISSUE TITLE
Implement browser caching for Recipients and Administrators

### ISSUE THREAD

**Issue by evilaliv3** [globaleaks/GlobaLeaks#1876]
It would be interesting to enable browser caching for Recipients and Administrators.

This would be a valuable feature that could speed up the application loading. In fact multiple resources served by the application are big but do not change during time (e.g. the javascript file is 2MB)

With the current state of analysis this would be easy doable by setting a cookie on the authenticated user of role admin and recipient.

This would potentially impact anonymous administrators and recipients so that it has been evaluated to enable such a capability only via HTTPS connections and not via Tor connections.

This ticket is to involve users of the platform in order to collect feedback and discuss the implementation.

**Comment by evilaliv3** [comment]
@Taipo i'm a lot interested in what you think about this ticket for your peculiar use case scenarios.

@filternetz @marceloomens maybe you have some comments as well?

**Comment by marceloomens** [comment]
@evilaliv3 In our case it would be fine to implement this for admins and recipients (but no cookie-setting for whistleblowers, off course!) I'm not sure if it's worth your effort though. We demand that admins and recipients access the platform using the Tor Browser Bundle (unless off course they're running their own Tor daemon). Isn't the Tor Browser Bundle configured to purge cookies and browser caches in between sessions by default?

**Comment by evilaliv3** [comment]
Yes exactly the Tor browser handle this deletion at closure but there is no way to detect that the user is using the Tor browser so that in relation to caching we can eventually evaluate to enable it only for Administrators/Recipients using Tor2web and this as discussed could be done only setting a cookie for thos users as soon that they will log in.

**Comment by evilaliv3** [comment]
@fpietrosanti what do you think in general about this? if applied only to admin and recipients using HTTPS i do not think it would impact what told inside the threat model.

in addiotion also crawlers could benefit from caching and would rank better the websites and make them be found more  so that it would be worth leaving caching open for them.

**Comment by fpietrosanti** [comment]
No security problem for "non-whistleblower" users to enable caching
